pkg/operator/apiservercontrollerset/apiservercontrollerset.go

```diff
@@ -0,0 +1,158 @@
+package apiservercontrollerset
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/apiservicecontroller"
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/nsfinalizercontroller"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	"github.com/openshift/library-go/pkg/operator/revisioncontroller"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/unsupportedconfigoverridescontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
+	apiregistrationinformers "k8s.io/kube-aggregator/pkg/client/informers/externalversions"
+)
+
+type APIServerControllerSet struct {
+	operatorClient v1helpers.OperatorClient
+	eventRecorder  events.Recorder
+
+	apiServiceController            *apiservicecontroller.APIServiceController
+	clusterOperatorStatusController *status.StatusSyncer
+	configUpgradableController      *unsupportedconfigoverridescontroller.UnsupportedConfigOverridesController
+	logLevelController              *loglevel.LogLevelController
+	finalizerController             *nsfinalizercontroller.FinalizerController
+	revisionController              *revisioncontroller.RevisionController
+}
+
+func NewAPIServerControllerSet(
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder,
+) *APIServerControllerSet {
+	apiServerControllerSet := &APIServerControllerSet{
+		operatorClient: operatorClient,
+		eventRecorder:  eventRecorder,
+	}
+
+	return apiServerControllerSet
+}
+
+// WithConfigUpgradableController adds a controller for the operator to check for presence of
+// unsupported configuration and to set the Upgradable condition to false if it finds any
+func (cs *APIServerControllerSet) WithConfigUpgradableController() *APIServerControllerSet {
+	cs.configUpgradableController = unsupportedconfigoverridescontroller.NewUnsupportedConfigOverridesController(cs.operatorClient, cs.eventRecorder)
+	return cs
+}
+
+func (cs *APIServerControllerSet) WithoutConfigUpgradableController() *APIServerControllerSet {
```

Comment:
do we really need these Without funcs?

Reply:
I see where this is going (https://github.com/openshift/cluster-openshift-apiserver-operator/pull/290/files#r365360059). Then we need a special value (e.g. the zero value for the controller structs) to mark `configUpgradableController` to be configured. Or maybe easier: add a string set of set up controllers and compare in the `PrepareRun`-like method (maybe called `Finalize`?).

Reply:
this may now make just a little bit more sense